donut/modules/courses/utils/import_registrar.py

```diff
@@ -36,135 +36,195 @@
-                    'course': course,
-                    'section_number': section,
-                    'instructor': instructor,
-                    'grades_type': grades_type,
-                    'times': times,
-                    'locations': locations
-                }
-        except e:
-            print('Error parsing section:', row_data, file=sys.stderr)
-            print(e, file=sys.stderr)
-
-with db.cursor() as cursor:
-    course_ids = {}
-    query = """
+                raise e
+
+
+def add_courses(year, term, courses, cursor):
+    insert_course_query = """
         INSERT INTO courses (
             year, term,
             department, course_number,
             name,
             units_lecture, units_lab, units_homework
         ) VALUES (%s, %s, %s, %s, %s, %s, %s, %s)
-        ON DUPLICATE KEY UPDATE course_id = course_id
+        ON DUPLICATE KEY UPDATE
+            name = VALUES(name),
+            units_lecture = VALUES(units_lecture),
+            units_lab = VALUES(units_lab),
+            units_homework = VALUES(units_homework)
+    """
+    get_course_query = """
+        SELECT course_id FROM courses
+        WHERE year = %s AND term = %s AND department = %s AND course_number = %s
     """
+    course_ids = {}
     for number, course in courses.items():
-        cursor.execute(
-            query, (year, term, course['department'], course['course_number'],
-                    course['name'], course['units_lecture'],
-                    course['units_lab'], course['units_homework']))
-        if cursor.lastrowid:  # 0 if no insert occured
-            course_ids[number] = cursor.lastrowid
-
-    def add_possibilities(name, values):
-        ids = {}
-        for value in values:
-            cursor.execute('SELECT ' + name + '_id AS id FROM ' + name +
-                           's WHERE ' + name + ' = %s LIMIT 1', value)
-            value_id = cursor.fetchone()
-            if value_id:
-                ids[value] = value_id['id']
-            else:
-                cursor.execute(
-                    'INSERT INTO ' + name + 's (' + name + ') VALUES (%s)',
-                    value)
-                ids[value] = cursor.lastrowid
-        return ids
-
-    instructor_ids = add_possibilities('instructor', instructors)
-    grades_types_ids = add_possibilities('grades_type', grades_types)
+        course_key = year, term, course['department'], course['course_number']
+        cursor.execute(insert_course_query, course_key +
+                       (course['name'], course['units_lecture'],
+                        course['units_lab'], course['units_homework']))
+        course_id = cursor.lastrowid
+        if course_id:
+            course_ids[number] = course_id
+            print('New course:', number)
+        else:
+            cursor.execute(get_course_query, course_key)
+            course_ids[number] = cursor.fetchone()['course_id']
+            print('Updated course:', number)
```

Comment:
remove the print?

Reply:
I added the prints intentionally so whoever runs the script has a sanity check that they imported the right file